### PR TITLE
yamlpath: init at 3.6.3

### PR DESCRIPTION
**Pull request by Flakebi**
###### Motivation for this change
yamlpath is a cli tool to work with yaml: https://github.com/wwkimball/yamlpath

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/yamlpath/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, fetchPypi
+, isPy3k
+, ruamel-yaml
+}:
+let
+  pinned-ruamel-yaml = ruamel-yaml.overridePythonAttrs (
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/yamlpath/default.nix:9]: This is not allowed for packages that are under pythonPackages.

File: pkgs/development/python-modules/yamlpath/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  src = fetchFromGitHub {
+    owner = "wwkimball";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "4lLKMMsjVWbnfiaOzdBePOtOwPN8nui3Ux6e55YdGoo=";
+  };
+
+  disabled = !isPy3k;
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/yamlpath/default.nix:32]:
```suggestion
```

File: pkgs/development/python-modules/yamlpath/default.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      src = fetchPypi {
+        inherit pname version;
+        sha256 = "EGvI1txqD/fJGWpHVwQyA29B1Va3eca05hgIX1fjnmc=";
+      };
+    }
+  );
+in
+buildPythonPackage rec {
+  pname = "yamlpath";
+  version = "3.6.3";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/yamlpath/default.nix:22]:
```suggestion
  version = "3.6.3";
  disabled = pythonOlder "3.6";
```

File: pkgs/development/python-modules/yamlpath/default.nix [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "4lLKMMsjVWbnfiaOzdBePOtOwPN8nui3Ux6e55YdGoo=";
+  };
+
+  disabled = !isPy3k;
+
+  propagatedBuildInputs = [ pinned-ruamel-yaml ];
+
+  # Tests need more dependencies that are not in nixpkgs
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/yamlpath/default.nix:35]: Please list them.

File: pkgs/development/python-modules/yamlpath/default.nix [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  doCheck = false;
+
+  pythonImportCheck = [ "yamlpath" ];
+
+  meta = with lib; {
+    homepage = "https://github.com/wwkimball/yamlpath";
+    description = "Command-line get/set/merge/validate/scan/convert/diff processors for YAML/JSON/Compatible data using powerful, intuitive, command-line friendly syntax";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/yamlpath/default.nix:42]:
```suggestion
   description = "Command-line processors for YAML/JSON/Compatible data";
    longDescription = ''
      Command-line get/set/merge/validate/scan/convert/diff processors for YAML/JSON/Compatible data using powerful, intuitive, command-line friendly syntax
     '';
```

File: pkgs/development/python-modules/yamlpath/default.nix (hunk 1 above)
Comment (Flakebi) [reply pkgs/development/python-modules/yamlpath/default.nix:9]: It doesn’t work with the version in nixpkgs (which is newer). What do you suggest to do instead?
`overridePythonAttrs` is already used for 6 packages inside python-modules.

File: pkgs/development/python-modules/yamlpath/default.nix (hunk 4 above)
Comment (Flakebi) [reply pkgs/development/python-modules/yamlpath/default.nix]: I found all of them now and got the tests working :)

**Comment by Flakebi** [comment]
Rebased

File: pkgs/development/python-modules/yamlpath/default.nix [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+  src = fetchFromGitHub {
+    owner = "wwkimball";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "4lLKMMsjVWbnfiaOzdBePOtOwPN8nui3Ux6e55YdGoo=";
+  };
+
+  propagatedBuildInputs = [ pinned-ruamel-yaml ];
+  checkInputs = [ hiera-eyaml mock pytest-console-scripts pytest-cov pytestCheckHook ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/yamlpath/default.nix:38]:
```suggestion
  checkInputs = [ hiera-eyaml mock pytest-console-scripts pytestCheckHook ];
```
We don't care about coverage

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/yamlpath/default.nix (hunk 1 above)
Comment (Flakebi) [reply pkgs/development/python-modules/yamlpath/default.nix]: Thanks for the review, I moved the package out of python-modules.

File: pkgs/development/tools/yamlpath/default.nix [hunk 7]
```diff
@@ -0,0 +1,45 @@
+{ lib
+, python3Packages
+, fetchFromGitHub
+, hiera-eyaml
+}:
+let
+  pinned-ruamel-yaml = python3Packages.ruamel-yaml.overridePythonAttrs (
```
Comment (SuperSandro2000) [inline pkgs/development/tools/yamlpath/default.nix:7]: Please use https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#overriding-python-packages-overriding-python-packages

**Pull request merged**